### ISSUE TITLE
train: codingjepa.training.dataloader (workers, deterministic seeding, intent-balanced sampling)

### ISSUE THREAD

**Issue by AbdelStark** [AbdelStark/CodingJEPA#68]
## Context

RFC-0008 §D5: workers=8, pin_memory=True, prefetch_factor=4, persistent_workers=True. Pretrain uses uniform random sampling; fine-tune uses intent-balanced sampling with class weights.

## Scope

- `codingjepa.training.dataloader.PretrainLoader(seq_parquet, batch, workers)` — uniform random.
- `codingjepa.training.dataloader.FinetuneLoader(pairs_parquet, batch, workers)` — intent-balanced sampler.
- Deterministic seeding via `worker_init_fn` and `torch.Generator`.

## Out of Scope

Sequence assembly (already offline; covered by #54).

## Design Reference

- RFC: [RFC-0008 §D5 §D9](../blob/main/docs/rfcs/RFC-0008-training-recipe-and-pipeline.md)

## Acceptance Criteria

- [ ] Two runs with the same seed produce identical batch order.
- [ ] Intent counts in fine-tune minibatches within ±10% of uniform.

## Dependencies

- Depends on: #54, #57

## Metadata

```
type: feature
area: training
priority: p0
effort: m
spec: rfc-0008
milestone: v0.2
tracking: #6
```
